WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -198,12 +200,11 @@ extension PointOfSaleAggregateModel {
 
     @MainActor
     func sendReceipt(to emailAddress: String) async {
-        guard let order = orderController.order else {
-            return
-        }
         // TODO:
         // Add eligiblity for correct WC and WCPay versions
-        await orderController.sendOrderReceipt(order: order, toEmailAddress: emailAddress)
+        if case let .loaded(_, order) = internalOrderState {
+            await orderController.sendOrderReceipt(order: order, recipientEmail: emailAddress)
+        }
```

Comment:
Since you're doing this on the `orderController`, you don't need to observe the `internalOrderState` again in the aggregate model; just make this function:

```suggestion
        await orderController.sendReceipt(recipientEmail: emailAddress)
```

And get the order within the orderController, which is fine to access its internal state.

Reply:
I see what you mean now, modified on 8af330cb2d37a05be4682252b30ab3bfb381e5e8